src/kryptos/k4/key_csp.py

```
@@ -120,15 +124,15 @@ def complete_partial_key(partial_key: list[int | None], ciphertext: str = K4) ->
 
-    known_slots = {i: s for i, s in enumerate(partial_key) if s is not None}
     unknown_slots = [i for i, s in enumerate(partial_key) if s is None]
     L = len(partial_key)
     ct = [c for c in ciphertext.upper() if c.isalpha()]
     eureka_words = {"EAST", "NORTHEAST", "BERLIN", "CLOCK"}
 
     # If all slots known, decrypt directly
     if not unknown_slots:
-        candidate = "".join(STANDARD[(STANDARD.index(c) - partial_key[i % L]) % 26] for i, c in enumerate(ct))
+        full_key: list[int] = [s for s in partial_key if s is not None]
+        candidate = "".join(STANDARD[(STANDARD.index(c) - full_key[i % L]) % 26] for i, c in enumerate(ct))
```

Pylint (4.0.7)

[convention] Line too long (107/88)

(C0301)